cloudcontainer.go

```diff
@@ -8,11 +8,18 @@ import (
 	"github.com/juju/schema"
 )
 
+// CloudContainer represents the state of a CAAS container, eg pod.
+type CloudContainer interface {
+	ProviderId() string
+	Address() Address
+	Ports() []string
+}
+
 type cloudContainer struct {
 	Version int `yaml:"version"`
 
 	ProviderId_ string   `yaml:"provider-id,omitempty"`
-	Address_    string   `yaml:"address,omitempty"`
+	Address_    *address `yaml:"address,omitempty"`
```

Comment:
Why is this a *address instead of just address ? Is it allowed to not be there?

Reply:
It's optional as we need to wait for k8s to notify us of the container address. So at the time of export the address may not be populated yet.